Gems/MotionMatching/Code/Source/KdTree.cpp

```diff
@@ -83,9 +83,29 @@ namespace EMotionFX::MotionMatching
         m_maxDepth = maxDepth;
         m_minFramesPerLeaf = minFramesPerLeaf;
 
+        // Not all features are present in the KD-tree, thus we need to remap KD-tree local feature columns to the
+        // feature schema global feature columns.
+        AZStd::vector<size_t> localToSchemaFeatureColumns;
+        {
+            localToSchemaFeatureColumns.resize(m_numDimensions);
+
+            size_t currentColumn = 0;
+            for (const Feature* feature : features)
+            {
+                const size_t numDimensions = feature->GetNumDimensions();
+                const size_t featureColumnOffset = feature->GetColumnOffset();
+                for (size_t i = 0; i < numDimensions; ++i)
+                {
+                    localToSchemaFeatureColumns[currentColumn] = featureColumnOffset + i;
+                    currentColumn++;
+                }
+            }
+
+            AZ_Assert(m_numDimensions == currentColumn, "There should be a column index mapping for each of the available dimensions.");
+        }
```

Comment:
question: I see this chunk of code is already in its own block, might it be worth extracting this to a function? 🙂 Something like `CalculateLocalToSchemaFeatureColumns(features, ...)`?

Reply:
Changed

Reply:
Awesome thanks!